x-pack/plugins/stack_connectors/server/connector_types/gemini/gemini.test.ts

```diff
@@ -0,0 +1,240 @@
+import { Logger } from '@kbn/logging';
+import type { Services } from '@kbn/actions-plugin/server/types';
+import {
+  KibanaRequest,
+  SavedObjectsClientContract,
+  ElasticsearchClient,
+  KibanaRequestEvents,
+  IKibanaSocket,
+} from '@kbn/core/server';
+import { ConnectorTokenClient } from '@kbn/actions-plugin/server/lib/connector_token_client';
+import type { Headers } from 'packages/core/http/core-http-server/src/router/headers';
+import { AxiosResponse, AxiosHeaders } from 'axios';
+import { RunApiResponse } from '../../../common/gemini/types';
+import { RunApiResponseSchema } from '../../../common/gemini/schema';
+
+jest.mock('@kbn/actions-plugin/server/sub_action_framework/helpers/validators', () => ({
+  assertURL: jest.fn(),
+}));
+
+jest.mock('@kbn/actions-plugin/server', () => {
+  const originalModule = jest.requireActual('@kbn/actions-plugin/server');
+
+  return {
+    ...originalModule,
+    SubActionConnector: jest.fn().mockImplementation(
+      (
+        ...args: ConstructorParameters<typeof originalModule.SubActionConnector> // Capture constructor arguments
+      ) => {
+        const connectorInstance = new originalModule.SubActionConnector(...args);
+        // Now you can mock the request method on the instance
+        (connectorInstance.request as jest.MockedFunction<
+          typeof originalModule.SubActionConnector.prototype.request
+        >) = jest.fn();
+        return connectorInstance;
+      }
+    ),
+  };
+});
+
+describe('GeminiConnector', () => {
+  let connector: GeminiConnector;
+  let mockServiceParams: ServiceParams<Config, Secrets>;
+
+  const mockConfigurationUtilities: ActionsConfigurationUtilities = {
+    isHostnameAllowed: jest.fn().mockReturnValue(true), // Always allow hostname
+    isUriAllowed: jest.fn().mockReturnValue(true), // Always allow URI
+    isActionTypeEnabled: jest.fn().mockReturnValue(true), // Always enable action type
+    ensureHostnameAllowed: jest.fn(), // Empty implementation for now
+    ensureUriAllowed: jest.fn(), // Empty implementation for now
+    ensureActionTypeEnabled: jest.fn(), // Empty implementation for now
+    getSSLSettings: jest.fn(), // Empty implementation for now
+    getProxySettings: jest.fn(), // Empty implementation for now
+    getResponseSettings: jest.fn(), // Empty implementation for now
+    getCustomHostSettings: jest.fn(), // Empty implementation for now
+    getMicrosoftGraphApiUrl: jest.fn().mockReturnValue('https://graph.microsoft.com/v1.0'),
+    getMicrosoftGraphApiScope: jest.fn().mockReturnValue('https://graph.microsoft.com/.default'),
+    getMicrosoftExchangeUrl: jest.fn().mockReturnValue('https://outlook.office.com/api/v2.0'),
+    getMaxAttempts: jest.fn().mockReturnValue(3),
+    validateEmailAddresses: jest.fn(),
+    enableFooterInEmail: jest.fn().mockReturnValue(true),
+    getMaxQueued: jest.fn().mockReturnValue(100),
+  };
+
+  const mockLogger: Logger = {
```

Comment:
Instead of defining a new mock, you can use the loggingSystemMock [as seen here in bedrock tests](https://github.com/elastic/kibana/blob/0fde2774a68cd449529ee615a3d46132c2097715/x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.test.ts#L76).

Reply:
cc: @shivankawasthi11

Reply:
@stephmilovic sure, will re-use the existing implementation